include/zephyr/net/sntp_server.h

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2025 Lothar Felten <lothar.felten@gmail.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief SNTP (Simple Network Time Protocol, RFC5905) Server
+ */
+
+#ifndef ZEPHYR_INCLUDE_NET_SNTP_SERVER_H_
+#define ZEPHYR_INCLUDE_NET_SNTP_SERVER_H_
+
+#include <zephyr/posix/sys/time.h>
```

Comment:
The zephyr internal header file should not depend to POSIX APIs. The sample etc can certainly include these if POSIX is enabled for it.